src/System.Net.Security/src/System/Net/Unix/SslConnectionInfo.cs

```diff
@@ -0,0 +1,101 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Security.Authentication;
+
+namespace System.Net
+{
+    internal class SslConnectionInfo
+    {
+        public readonly int Protocol;
+        public readonly int DataCipherAlg;
+        public readonly int DataKeySize = 0;
+        public readonly int DataHashAlg;
+        public readonly int DataHashKeySize = 0;
+        public readonly int KeyExchangeAlg;
+        public readonly int KeyExchKeySize = 0;
+
+        internal SslConnectionInfo(Interop.libssl.SSL_CIPHER cipher)
+        {
+            Protocol = (int) MapProtocol(cipher.algorithm_ssl);
```

Comment:
This is inline with windows version. Windows one gets an int from lower level Apis.